src/prefect/experimental/workers/base.py

```diff
@@ -0,0 +1,590 @@
+import abc
+from functools import partial
+from pathlib import Path
+from typing import Dict, List, Optional
+from uuid import uuid4
+
+import anyio
+import anyio.abc
+import pendulum
+from fastapi import FastAPI
+from pydantic import BaseModel, ValidationError
+
+from prefect._internal.compatibility.experimental import experimental
+from prefect.client.orion import OrionClient, get_client
+from prefect.client.schemas import FlowRun
+from prefect.deployments import Deployment
+from prefect.engine import propose_state
+from prefect.exceptions import Abort, ObjectNotFound
+from prefect.orion import schemas
+from prefect.orion.schemas.responses import WorkerFlowRunResponse
+from prefect.orion.services.loop_service import LoopService
+from prefect.settings import (
+    PREFECT_WORKER_HEARTBEAT_SECONDS,
+    PREFECT_WORKER_PREFETCH_SECONDS,
+    PREFECT_WORKER_QUERY_SECONDS,
+    PREFECT_WORKER_WORKFLOW_STORAGE_PATH,
+    PREFECT_WORKER_WORKFLOW_STORAGE_SCAN_SECONDS,
+)
+from prefect.states import Crashed, Pending, exception_to_failed_state
+from prefect.utilities.dispatch import register_base_type
+from prefect.utilities.services import critical_service_loop
+
+
+class BaseWorkerResult(BaseModel, abc.ABC):
+    identifier: str
+    status_code: int
+
+    def __bool__(self):
+        return self.status_code == 0
+
+
+@register_base_type
+class BaseWorker(LoopService, abc.ABC):
+    type: str
+    loop_seconds = PREFECT_WORKER_QUERY_SECONDS.value()
+    prefetch_seconds = PREFECT_WORKER_PREFETCH_SECONDS.value()
+    heartbeat_seconds = PREFECT_WORKER_HEARTBEAT_SECONDS.value()
+    workflow_storage_scan_seconds = PREFECT_WORKER_WORKFLOW_STORAGE_SCAN_SECONDS.value()
+    workflow_storage_path = PREFECT_WORKER_WORKFLOW_STORAGE_PATH.value()
```

Comment:
This is not a safe way to initialize values from settings since these are evaluated at module load time and settings can be context managed after that.

Reply:
Good catch! I moved the value initialization from settings into `__init__`, but let me know if there's a more clever way to do it.